user_sync/credential_manager.py

```diff
@@ -42,5 +46,22 @@ def get_credentials(credential_type, credential_id, **kwArgs):
     :type kwArgs: dict
     :rtype dict | str | None
     '''
+    if credential_type == DIRECTORY_CREDENTIAL_TYPE:
+        config = kwArgs['config']
+        if "credential_manager" in config:
+            cred_man = config['credential_manager']
+            for supported in KEYRING_SUPPORTED:
+                if cred_man['type'] == supported:
```

Comment:
If there is no 'type' element in cred_man (user forgot it), this will throw.  Use cred_man.get('type') which will return None if the key is missing.  Even better is to eliminate the loop, and just say
```
if cred_man.get('type') in KEYRING_SUPPORTED:
    return ...
```